### DISCUSSION TITLE
Buggy livewire loop on input

### DISCUSSION THREAD

**Discussion by kgcusi** [livewire/livewire#3458] [1. Help]
I have a looped input date in my livewire component, the loop works but every other request that I make to change the `$term` the loop runs again but shows 1 less input date.

My term model, I have a fake button to help the user remove focus on the input thus making the request to change the value.
```
<input class="form-control @error('term') is-invalid @enderror " id="term" type="number" name="term" wire:model.lazy="term" min='0' step='1'>
```

This is the loop.
```
@for ($i; $i <= $term-1; $i++) 
       <div class="form-group row ">
       <label for="date" class="col-md-3 col-form-label "><small>{{$i+1}}.</small> Date
       <span style="color: red">*</span></label>
       <div class="col-md-9">
          <input class="form-control @error('date') is-invalid @enderror" id="date" type="date" name="date" 
           wire:model.defer="date.{{$i}}">
          @error('date') <div class="invalid-feedback">{{ $message }}</div>
          @enderror
        </div>
        </div>
         @endfor
```
First request
![image](https://user-images.githubusercontent.com/69495865/127835251-51946d8a-fbc5-4550-bef7-076740ba6c53.png)

Second request
![image](https://user-images.githubusercontent.com/69495865/127835304-21c3b5cb-db12-41b9-9d67-84d96e206c27.png)

If I try to make another request to change `$term` it will work for e.g I put `6` it will show `6` input dates but if I try to change again it will not render one input date like the second request above.

**Comment by Qirel** [comment]
Your code, if that's all of it (given the picture, there appears to be more of it), violates some of Livewire's principles on rendering HTML.

1. Each view should have only 1 root element, which means that the entire view should be wrapped in one element, like a `<div>`. 
2. Items that is dynamically rendered in a loop, MUST have a unique `wire:key` on it, so Livewire's DOM-diffing can track each element.
3. Items that is dynamically rendered in a loop, must have a single root element in it, similar to point 1.

You also were generating multiple elements with the same ID on the HTML elements, that is also something to take note of. 

Some sensible formatting also makes the code easier to read (and in turn, troubleshoot). If you fix these things, the error will most likely go away, like this

```blade
<div>
    <input 
        class="form-control @error('term') is-invalid @enderror "
        id="term" 
        type="number" 
        name="term"
        wire:model.lazy="term" 
        min='0' 
        step='1'
    />

    @for ($i = 0; $i < $term; $i++) 
        <div class="form-group row" wire:key="dates.{{ $i }}">
            <label for="date-{{ $i }}" class="col-md-3 col-form-label">
                <small>{{ $i + 1 }}.</small> 
                Date
                <span style="color: red">*</span>
            </label>
            <div class="col-md-9">
                <input 
                    class="form-control @error('date') is-invalid @enderror" 
                    id="date-{{ $i }}" 
                    type="date" 
                    name="date"
                    wire:model.defer="date.{{ $i }}" 
                />
                @error('date') 
                    <div class="invalid-feedback">{{ $message }}</div>
                @enderror
            </div>
        </div>
    @endfor
</div>
```
